jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -765,9 +763,7 @@ public final boolean isAlive() {
  */
 private boolean isDead() {
 	// Has already started, is not alive anymore, and has been removed from the ThreadGroup
```

Comment:
This comment (and the doc comment) suggests we are doing some type of checking w.r.t. removal from a ThreadGroup. Do we know if `started` indicates anything about ThreadGroup removal? Either way, while we are here we should ensure that the comment is accurate.

Reply:
`threadRef` can be NULL before the thread starts and after it exits. `started = TRUE` makes sure that we only pick up the exit case and implies that thread has been removed from the `ThreadGroup`.

Reply:
What will be your suggestion for a rewrite?

Reply:
There is also a reference to a very old issue: PR 1FJMO7Q; I believe it is no longer accessible. They might have seen an issue related to `ThreadGroup` and `isDead` method.

May be, change it to the following to clarify and highlight the implicit relationship?
```suggestion
	/* threadRef can be NO_REF before the thread starts and after it exits.
	 * Only account for the exit case where threadRef is NO_REF. This assures
	 * that the thread has been removed from the ThreadGroup.
	 */ 
```

Reply:
> implies that thread has been removed from the ThreadGroup.

It was this implication that isn't clear to me. I'd've thought that:
 `// Had already started and is not alive anymore.`
 would alone be appropriate as the comment.
 
The only place I could find where the thread is removed from the ThreadGroup is [here](https://github.com/eclipse-openj9/openj9/blob/6544c13e4e911528290f3d546024780f307e6e0e/jcl/src/java.base/share/classes/java/lang/Thread.java#L1150).

But that isn't through the `exit()` path. That is in the failure to `start()` path. Exit simply `null`s out the `group` member.

Reply:
Oh, I see. It looks like the code has changed; but, the comment was never updated. We can probably remove the reference to the old issue and `ThreadGroup`.

Reply:
That is what I am thinking too. But I am also wondering now if we are missing a call to `this.group.remove(this)` during the exit sequence.

Reply:
In OpenJ9's `ThreadGroup.java` (JDK17-), there is a comment which is repeated twice: `Dead Threads not removed from ThreadGroups`. There might have been a behavior change where dead threads were no longer removed from the `ThreadGroup`. Also, methods like `ThreadGroup.activeCount` are anticipating dead threads since they are doing checks like `isAlive`.

In JDK21+, `ThreadGroup` in the extension repo no longer maintains a list of threads; instead, it relies upon `Thread.getAllThreads()`.

`ThreadGroups` are not widely used and considered unsafe ([ref](https://stackoverflow.com/questions/18897621/why-is-not-safe-to-use-java-lang-threadgroup)). Since no functional issues are reported, it might not be worth investigating this issue.

Reply:
Sounds good.